src/box/box.cc

```diff
@@ -681,6 +681,157 @@ box_check_uri(const char *source, const char *option_name)
 	return 0;
 }
 
+static inline int
+box_check_uri_delimiters(const char *source)
+{
+	static const char *uri_delimiters = ", ?=&;\0";
+	static const uint32_t uri_delimiters_size = 7;
+	size_t source_len = strlen(source);
+	source_len = (source_len != 0 ? source_len : source_len + 1);
+
+	for (uint32_t i = 0; i < uri_delimiters_size; i++) {
+		if (source[0] == uri_delimiters[i] ||
+		    source[source_len - 1] == uri_delimiters[i]) {
+			diag_set(ClientError, ER_CFG, "listen",
+				 "expected host:service or /unix.socket");
+			return -1;
+		}
+	}
+	return 0;
+}
+
+static inline int
+box_check_end_of_token(const char *delimiter, char **saveptr,
+		       const char *error)
+{
+	if (strtok_r(NULL, delimiter, saveptr) != NULL) {
+		diag_set(ClientError, ER_CFG, "listen", error);
+		return -1;
+	}
+	return 0;
+}
+
+static inline int
+box_check_uri_option(const char *opt, const char *val)
```

Comment:
I would design it right know the way which would be easy to expand with other option/values.
Like:
enum uri_option_key {
	URI_OPTION_KEY_TRANPORT = 0,
	URI_OPTION_KEY_UNKNOWN
};

enum uri_option_value {
	URI_OPTION_VALUE_PLAIN = 0,
	URI_OPTION_VALUE_UNKNOWN
};

enum uri_option_key
parse_uri_option_key(const char *opt)
{
	if (strncmp(opt, "transport", strlen("transport")) == 0)
		return URI_OPTION_KEY_TRANPORT;
	return URI_OPTION_KEY_UNKNOWN;
}

enum uri_option_value
parse_uri_option_value()

static inline int
box_check_uri_option(const char *opt, const char *val)
{
	enum uri_option_key key = parse_uri_option_key(opt);
	if (key == URI_OPTION_KEY_UNKNOWN) {
		diag_set(ClientError, ER_CFG, "listen",
			 tt_sprintf("invalid option `%s` for URI", opt));
		return -1;
	}
	enum uri_option_value value = parse_uri_option_value(val);
	if (value == URI_OPTION_KEY_UNKNOWN) {
		diag_set(ClientError, ER_CFG, "listen",
			 tt_sprintf("invalid value `%s` for "
				    "`transport` option", val));
		return -1;
	}
	return 0;
}

Reply:
Fixed